<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'>
<head>
<title>poky - Poky Build Tool and Metadata</title>
<meta name='generator' content='cgit v0.9.1'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit/cgit.css'/>
<link rel='shortcut icon' href='/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='http://git.yoctoproject.org/cgit/cgit.cgi/poky/atom/meta/recipes-graphics/wayland/weston/groups.patch?h=master' type='application/atom+xml'/>
</head>
<body>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/cgit/cgit.cgi/'><img src='/cgit/yocto-project-transp.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/cgit/cgit.cgi/'>index</a> : <a title='poky' href='/cgit/cgit.cgi/poky/'>poky</a></td><td class='form'><form method='get' action=''>
<select name='h' onchange='this.form.submit();'>
<option value='1.1_M1'>1.1_M1</option>
<option value='1.1_M2'>1.1_M2</option>
<option value='1.1_M3'>1.1_M3</option>
<option value='1.1_M4'>1.1_M4</option>
<option value='1.2_M1'>1.2_M1</option>
<option value='1.2_M2'>1.2_M2</option>
<option value='1.2_M3'>1.2_M3</option>
<option value='1.2_M4'>1.2_M4</option>
<option value='1.3_M1'>1.3_M1</option>
<option value='1.3_M2'>1.3_M2</option>
<option value='1.3_M3'>1.3_M3</option>
<option value='1.3_M4'>1.3_M4</option>
<option value='1.3_M5'>1.3_M5</option>
<option value='1.3_beta'>1.3_beta</option>
<option value='1.4_M1'>1.4_M1</option>
<option value='1.4_M3'>1.4_M3</option>
<option value='1.4_M4'>1.4_M4</option>
<option value='1.4_M5'>1.4_M5</option>
<option value='bernard'>bernard</option>
<option value='blinky'>blinky</option>
<option value='clyde'>clyde</option>
<option value='danny'>danny</option>
<option value='danny-next'>danny-next</option>
<option value='denzil'>denzil</option>
<option value='dylan'>dylan</option>
<option value='edison'>edison</option>
<option value='elroy'>elroy</option>
<option value='green'>green</option>
<option value='laverne'>laverne</option>
<option value='master' selected='selected'>master</option>
<option value='master-next'>master-next</option>
<option value='pinky'>pinky</option>
<option value='purple'>purple</option>
</select> <input type='submit' name='' value='switch'/></form></td></tr>
<tr><td class='sub'>Poky Build Tool and Metadata</td><td class='sub right'>git repository hosting</td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/cgit.cgi/poky/'>summary</a><a href='/cgit/cgit.cgi/poky/refs/'>refs</a><a href='/cgit/cgit.cgi/poky/log/meta/recipes-graphics/wayland/weston/groups.patch'>log</a><a class='active' href='/cgit/cgit.cgi/poky/tree/meta/recipes-graphics/wayland/weston/groups.patch'>tree</a><a href='/cgit/cgit.cgi/poky/commit/meta/recipes-graphics/wayland/weston/groups.patch'>commit</a><a href='/cgit/cgit.cgi/poky/diff/meta/recipes-graphics/wayland/weston/groups.patch'>diff</a><a href='/cgit/cgit.cgi/poky/stats/meta/recipes-graphics/wayland/weston/groups.patch'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/cgit.cgi/poky/log/meta/recipes-graphics/wayland/weston/groups.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/cgit.cgi/poky/tree/'>root</a>/<a href='/cgit/cgit.cgi/poky/tree/meta'>meta</a>/<a href='/cgit/cgit.cgi/poky/tree/meta/recipes-graphics'>recipes-graphics</a>/<a href='/cgit/cgit.cgi/poky/tree/meta/recipes-graphics/wayland'>wayland</a>/<a href='/cgit/cgit.cgi/poky/tree/meta/recipes-graphics/wayland/weston'>weston</a>/<a href='/cgit/cgit.cgi/poky/tree/meta/recipes-graphics/wayland/weston/groups.patch'>groups.patch</a></div><div class='content'>blob: 9dc043ac3bca4cbd63eeea3b6f411cffb5dbfa19 (<a href='/cgit/cgit.cgi/poky/plain/meta/recipes-graphics/wayland/weston/groups.patch'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a class='no' id='n1' name='n1' href='#n1'>1</a>
<a class='no' id='n2' name='n2' href='#n2'>2</a>
<a class='no' id='n3' name='n3' href='#n3'>3</a>
<a class='no' id='n4' name='n4' href='#n4'>4</a>
<a class='no' id='n5' name='n5' href='#n5'>5</a>
<a class='no' id='n6' name='n6' href='#n6'>6</a>
<a class='no' id='n7' name='n7' href='#n7'>7</a>
<a class='no' id='n8' name='n8' href='#n8'>8</a>
<a class='no' id='n9' name='n9' href='#n9'>9</a>
<a class='no' id='n10' name='n10' href='#n10'>10</a>
<a class='no' id='n11' name='n11' href='#n11'>11</a>
<a class='no' id='n12' name='n12' href='#n12'>12</a>
<a class='no' id='n13' name='n13' href='#n13'>13</a>
<a class='no' id='n14' name='n14' href='#n14'>14</a>
<a class='no' id='n15' name='n15' href='#n15'>15</a>
<a class='no' id='n16' name='n16' href='#n16'>16</a>
<a class='no' id='n17' name='n17' href='#n17'>17</a>
<a class='no' id='n18' name='n18' href='#n18'>18</a>
<a class='no' id='n19' name='n19' href='#n19'>19</a>
<a class='no' id='n20' name='n20' href='#n20'>20</a>
<a class='no' id='n21' name='n21' href='#n21'>21</a>
<a class='no' id='n22' name='n22' href='#n22'>22</a>
<a class='no' id='n23' name='n23' href='#n23'>23</a>
<a class='no' id='n24' name='n24' href='#n24'>24</a>
<a class='no' id='n25' name='n25' href='#n25'>25</a>
<a class='no' id='n26' name='n26' href='#n26'>26</a>
<a class='no' id='n27' name='n27' href='#n27'>27</a>
<a class='no' id='n28' name='n28' href='#n28'>28</a>
<a class='no' id='n29' name='n29' href='#n29'>29</a>
<a class='no' id='n30' name='n30' href='#n30'>30</a>
<a class='no' id='n31' name='n31' href='#n31'>31</a>
<a class='no' id='n32' name='n32' href='#n32'>32</a>
<a class='no' id='n33' name='n33' href='#n33'>33</a>
<a class='no' id='n34' name='n34' href='#n34'>34</a>
<a class='no' id='n35' name='n35' href='#n35'>35</a>
<a class='no' id='n36' name='n36' href='#n36'>36</a>
<a class='no' id='n37' name='n37' href='#n37'>37</a>
<a class='no' id='n38' name='n38' href='#n38'>38</a>
<a class='no' id='n39' name='n39' href='#n39'>39</a>
<a class='no' id='n40' name='n40' href='#n40'>40</a>
<a class='no' id='n41' name='n41' href='#n41'>41</a>
<a class='no' id='n42' name='n42' href='#n42'>42</a>
<a class='no' id='n43' name='n43' href='#n43'>43</a>
<a class='no' id='n44' name='n44' href='#n44'>44</a>
<a class='no' id='n45' name='n45' href='#n45'>45</a>
<a class='no' id='n46' name='n46' href='#n46'>46</a>
<a class='no' id='n47' name='n47' href='#n47'>47</a>
</pre></td>
<td class='lines'><pre><code>Upstream-Status: Submitted (https://bugs.freedesktop.org/show_bug.cgi?id=65933)
Signed-off-by: Ross Burton &lt;ross.burton@intel.com&gt;

From 42821739a228a85ce3432be1796858e5cc31688b Mon Sep 17 00:00:00 2001
From: Quentin Glidic &lt;sardemff7+git@sardemff7.net&gt;
Date: Wed, 19 Jun 2013 15:27:11 +0200
Subject: [PATCH weston] weston-launch: Set all groups for user

Signed-off-by: Quentin Glidic &lt;sardemff7+git@sardemff7.net&gt;
---
 configure.ac        | 2 +-
 src/weston-launch.c | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b625221..db3feb0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,7 +52,7 @@ AC_CHECK_DECL(CLOCK_MONOTONIC,[],
 	      [[#include &lt;time.h&gt;]])
 AC_CHECK_HEADERS([execinfo.h])
 
-AC_CHECK_FUNCS([mkostemp strchrnul])
+AC_CHECK_FUNCS([mkostemp strchrnul initgroups])
 
 COMPOSITOR_MODULES="wayland-server &gt;= 1.1.90 xkbcommon pixman-1"
 
diff --git a/src/weston-launch.c b/src/weston-launch.c
index 76dcede..7d7b556 100644
--- a/src/weston-launch.c
+++ b/src/weston-launch.c
@@ -631,9 +631,13 @@ main(int argc, char *argv[])
 		}
 
 		if (setgid(wl.pw-&gt;pw_gid) &lt; 0 ||
+#ifdef HAVE_INITGROUPS
+                    initgroups(wl.pw-&gt;pw_name, wl.pw-&gt;pw_gid) &lt; 0 ||
+#endif
 		    setuid(wl.pw-&gt;pw_uid) &lt; 0)
 			error(1, errno, "dropping privilidges failed");
 
+
 		if (sleep_fork) {
 			if (wl.verbose)
 				printf("weston-launch: waiting %d seconds\n", sleep_fork);
-- 
1.8.3
</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class='footer'>generated  by cgit v0.9.1 at 2013-09-17 12:41:12 (GMT)</div>
</div> <!-- id=cgit -->
</body>
</html>
